src/shared/components/search-button/SearchButton.tsx

```diff
@@ -14,29 +14,29 @@
  * limitations under the License.
  */
 
-import { createSlice, type PayloadAction } from '@reduxjs/toolkit';
+import classNames from 'classnames';
 
-type SpeciesSelectorGeneSearchState = {
-  query: string;
-};
+import SearchIcon from 'static/icons/icon_search.svg';
+
+import styles from './SearchButton.module.css';
 
-const initialState: SpeciesSelectorGeneSearchState = {
-  query: ''
+type Props = {
+  onClick: () => void;
+  label: string;
+  className?: string;
 };
```

Comment:
If `SearchButton` becomes a very generic component:

- Does it always need a label as a prop?
- If we decide that it may exist without a label, should it in that case be able to have an aria-label for screen readers?
- Could you add it to Storybook?